spec.html

```diff
@@ -31678,20 +31538,167 @@
+        <emu-alg>
+          1. Assert: Type(_buffer_) is Object and _buffer_ has an [[ArrayBufferData]] internal slot.
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. Let _O_ be ? AllocateTypedArray(_TypedArray_).
+          1. Let _constructorName_ be the String value of _O_'s [[TypedArrayName]] internal slot.
+          1. Let _elementSize_ be the Number value of the Element Size value in <emu-xref href="#table-49"></emu-xref> for _constructorName_.
+          1. Let _offset_ be ? ToInteger(_byteOffset_).
+          1. If _offset_ &lt; 0, throw a *RangeError* exception.
+          1. If _offset_ is -0, let _offset_ be +0.
+          1. If _offset_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+          1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
+          1. Let _bufferByteLength_ be the value of _buffer_'s [[ArrayBufferByteLength]] internal slot.
+          1. If _length_ is *undefined*, then
+            1. If _bufferByteLength_ modulo _elementSize_ &ne; 0, throw a *RangeError* exception.
+            1. Let _newByteLength_ be _bufferByteLength_ - _offset_.
+            1. If _newByteLength_ &lt; 0, throw a *RangeError* exception.
+          1. Else,
+            1. Let _newLength_ be ? ToLength(_length_).
+            1. Let _newByteLength_ be _newLength_ &times; _elementSize_.
+            1. If _offset_+_newByteLength_ &gt; _bufferByteLength_, throw a *RangeError* exception.
+          1. Set _O_'s [[ViewedArrayBuffer]] internal slot to _buffer_.
+          1. Set _O_'s [[ByteLength]] internal slot to _newByteLength_.
+          1. Set _O_'s [[ByteOffset]] internal slot to _offset_.
+          1. Set _O_'s [[ArrayLength]] internal slot to _newByteLength_ / _elementSize_ .
+          1. Return _O_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="typedarray-create">
+        <h1>TypedArrayCreate( _constructor_, _argumentList_ )</h1>
+        <emu-alg>
+          1. Let _newTypedArray_ be ? Construct(_constructor_, _argumentList_).
+          1. Perform ? ValidateTypedArray(_newTypedArray_).
+          1. If _argumentList_ is a List of a single Number,
+            1. If the value of _newTypedArray_'s [[ArrayLength]] internal slot < _argumentList_[0], throw a *TypeError* exception.
+          1. Return _newTypedArray_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="typedarray-species-create">
+        <h1>TypedArraySpeciesCreate( _exemplar_, _argumentList_ )</h1>
+        <emu-alg>
+          1. Let _defaultConstructor_ be the intrinsic object listed in column one of <emu-xref href="#table-49"></emu-xref> for the value of _exemplar_'s [[TypedArrayName]] internal slot.
+          1. Let _constructor_ be ? SpeciesConstructor(_exemplar_, _defaultConstructor_).
+          1. Return ? TypedArrayCreate(_constructor_, _argumentList_).
```

Comment:
Can you create an issue to add `?` in Return positions, so it's used consistently and not just in TypedArraySpeciesCreate? Thanks in advance! :smile: